pkg/utils/utils.go

```diff
@@ -431,3 +438,51 @@ func getLinkType(ifaceStatus sriovnetworkv1.InterfaceExt) string {
 
 	return ""
 }
+
+func setVfsGuid(iface *sriovnetworkv1.Interface) error {
+	glog.Infof("setVfsGuid(): device %s", iface.PciAddress)
+	pfLink, err := netlink.LinkByName(iface.Name)
+	if err != nil {
+		glog.Errorf("setVfsGuid(): unable to get PF link for device %+v %q", iface, err)
+		return err
+	}
+	vfs, err := dputils.GetVFList(iface.PciAddress)
+	if err != nil {
+		return err
+	}
+	for _, addr := range vfs {
+		vfID, err := dputils.GetVFID(addr)
+		if err != nil {
+			glog.Errorf("setVfsGuid(): unable to get VF id %+v %q", iface.PciAddress, err)
+			return err
+		}
+		guid := generateRandomGuid()
+		if err := netlink.LinkSetVfNodeGUID(pfLink, vfID, guid); err != nil {
+			return err
+		}
+		if err := netlink.LinkSetVfPortGUID(pfLink, vfID, guid); err != nil {
+			return err
+		}
+		if err = Unbind(addr); err != nil {
```

Comment:
please correct me if I'm wrong:
1) For RoCE VF device to get node or port guid, it needs to first set VF admin MAC, then unbind/bind driver
2) For IB or IPoIB VF device to get node or port guid, it needs to explicitly set the VF GUID , then unbind/bind driver

If the issue of `RDMA_CM cannot work without node or port guid` is fixed, then we can remove unbind/bind.

Reply:
It true, but for the IB we alway have to set port_guid  and unbind/bind the driver.
This is need for Infiniband to work not releated to rdma_cm
without ib-kubernetes we will do it in the operator. 
in the future if we decided that  ib-kubernetes is a must the we can remove it from here and let   ib-kubernete and ib-sriov-cni to set it.

Reply:
For IB SR-IOV, can we set a random default guid in ib-sriov-cni if not provided by user or ib-kubernetes?

Reply:
we can do it, but I thought to make it similar to RoCE so it will be one time change.

Reply:
ok, I think the initialization of node/port guid is needed for IB device where ib-sriov-cni might not be used. so it will not resolve the issue when adding in ib-sriov-cni for IB.